src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -843,39 +846,51 @@ private void StopTimers()
 
         private void CleanupCandidatePeers()
         {
-            if (_peerPool.PeerCount <= _networkConfig.CandidatePeerCountCleanupThreshold)
-            {
-                return;
-            }
+            int peerCount = _peerPool.PeerCount;
 
-            // may further optimize allocations here
-            List<Peer> candidates = _peerPool.NonStaticPeers;
-            int countToRemove = candidates.Count - _networkConfig.MaxCandidatePeerCount;
-            Peer[] failedValidationCandidates = candidates.Where(x => _stats.HasFailedValidation(x.Node))
-                .OrderBy(x => _stats.GetCurrentReputation(x.Node)).ToArray();
-            Peer[] otherCandidates = candidates.Except(failedValidationCandidates).Except(_peerPool.ActivePeers.Values).OrderBy(x => _stats.GetCurrentReputation(x.Node)).ToArray();
```

Comment:
_Except(_peerPool.ActivePeers.Values)_ condition has been removed. Are we ok with that?

Reply:
Good catch. I think I missed it! Those need to be sorted as last items and later removed from the collection!

This is very important for this PR!

Reply:
Could we add a unit test for that to avoid breaking the condition again in the future?